documentation/AuthServer.md

```diff
@@ -25,56 +53,110 @@ The document types used by the authorization server include:
 
-#### Client
+#### GPII Tokens
+
+| Option | Type | Description | Default |
+| ------ | ---- | ----------- | ------- |
+| `id` | String | The GPII token id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
+| `type` | String | The document type for storing GPII tokens. | The value must be set to "gpiiToken". |
+| `gpiiToken` | String | The GPII token. | None |
+| `userId` | String | The user id that this GPII token belongs to. | None |
+
+#### GPII App Installation Clients
 
 | Option | Type | Description | Default |
 | ------ | ---- | ----------- | ------- |
 | `id` | String | The client id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
-| `type` | String | The document type for storing client information. | The value must be set to "client". |
+| `type` | String | The document type for storing the client type information. | The value must be set to "gpiiAppInstallationClient". |
 | `name` | String | The client name. | None |
 | `oauth2ClientId` | String | The unique identifier issued to a registered OAuth2 client by the authorization server. | None |
 | `oauth2ClientSecret` | String | Confidential shared secret between the client and the authorization server, used to verify the identity of the client. | None |
-| `redirectUri` | String | The URL on client's site where users will be sent after authorization. | None |
+| `userId` | Boolean | The id of the user who creates this client.| None |
 
-#### GPII Tokens
+#### GPII App Installation Authorizations
 
 | Option | Type | Description | Default |
 | ------ | ---- | ----------- | ------- |
-| `id` | String | The GPII token id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
-| `type` | String | The document type for storing GPII tokens. | The value must be set to "gpiiToken". |
-| `gpiiToken` | String | The GPII token. | None |
-| `userId` | String | The user id that this GPII token belongs to. | None |
+| `id` | String | The GPII app installation authorizations id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
+| `type` | String | The document type for storing GPII app installation authorizationss. | The value must be set to "gpiiAppInstallationAuthorization". |
+| `clientId` | String | The client id that this token is assigned to. | None |
+| `gpiiToken` | String | The GPII token that this token record is associated with. | None |
+| `accessToken` | String | The access token used to retrieved the protected user preferences. | None |
+| `expiresIn` | Number | The number of seconds left before the access token becomes invalid. The default value is 3600. | 3600 |
+| `revoked` | Boolean | Whether this token has been revoked. | false |
+| `expired` | Boolean | Whether this token has expired, meaning `timestampCreated` + `expiresIn` < now(). | false |

```

Comment:
Sorry to confuse things. I've now done some research and I think we would accomplish this querying without calculating and storing `expired` ourselves by adding the hypothetical `timestampExpires` onto our view key as in:

```
"findGpiiAppInstallationAuthorizationByGpiiTokenAndClientId": {
    "map": "function(doc) {if (doc.type === 'gpiiAppInstallationAuthorization' && doc.revoked === false) {emit([doc.gpiiToken, doc.clientId, doc.timestampExpires], doc)}}"
}
```

And then used the ranged query `startkey` and `endkey`, to include the current time in the query, rather than `key`. As in something like:

```
findGpiiAppInstallationAuthorizationByGpiiTokenAndClientId?startKey=["%gpiiToken","%clientId", currentTime]&endkey=["%gpiiToken","%clientId", currentTime + a day]
```

Is is better than calculating and storing the expired state as you have done? I'm not sure.